src/clusterfuzz/_internal/base/tasks.py

```diff
@@ -247,40 +230,72 @@ def get_machine_templates():
   return conf['instance_templates']
 
 
-def _get_messages(pubsub_client, application_id, queue):
-  return pubsub_client.pull_from_subscription(
-      pubsub.subscription_name(application_id, queue), max_messages=1)
+class PubSubPuller:
+  """PubSub client providing convenience methods for pulling."""
+
+  def __init__(self, queue):
+    self.pubsub_client = pubsub.PubSubClient()
```

Comment:
nit: s/pubsub_client/client/, since it's obvious in the context of the `PubSubPuller` class what `client` is referring to.